uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -285,7 +291,14 @@ define([
             initAbsenceType();
             initStatus();
             initContact();
-            initDates();
+
+            //the promise will set the day types so that initial value is set properly
+            return initDates();
+          })
+          .then( function () {
+            if(vm.mode == 'edit') {
+              initialLeaveRequest = _.cloneDeep(vm.leaveRequest.attributes());
```

Comment:
Also I think the name of this variable is incorrect, as it's only containing the attributes of the leave request, not the leave request per se